### ISSUE TITLE
FR: Option to automatically add words from Obsidian#s built-in dictionary to the completion list

### ISSUE THREAD

**Issue by chrisgrieser** [tadashi-aikawa/obsidian-various-complements-plugin#55]
<img width="1090" alt="Screenshot 2021-12-23 19 41 06" src="https://user-images.githubusercontent.com/73286100/147280355-b4afb2fd-6d9d-4bbc-a602-8af882b39569.png">
<img width="270" alt="Screenshot 2021-12-23 19 42 23" src="https://user-images.githubusercontent.com/73286100/147280469-2a7b0a2b-75bb-44ab-add9-b15efdba1df8.png">

I noticed that quite often, the words that I manually add to Obsidian's dictionary are often long academic terms that I'd also want to have suggested for autocompletes.

Is it possible to have an option to add the words Obsidian saves in that "Ignored Words" list also to the list of words suggested? (They are saved in `.obsidian/app.json`, btw.)

**Comment by tadashi-aikawa** [comment]
Hi, @chrisgrieser. Thank you for your FR :)

> I noticed that quite often, the words that I manually add to Obsidian's dictionary are often long academic terms that I'd also want to have suggested for autocompletes.

Right. I think so, too.

> They are saved in .obsidian/app.json, btw.

I have added a word to the dictionary and seen `.obsidian/app.json`. But there are no words I added.

```json
{
  "autoPairBrackets": false,
  "showLineNumber": true,
  "vimMode": true,
  "promptDelete": false,
  "alwaysUpdateLinks": true,
  "newFileLocation": "folder",
  "newFileFolderPath": "Notes/2021-51w",
  "showUnsupportedFiles": true,
  "attachmentFolderPath": "./attachments",
  "autoConvertHtml": false,
  "autoPairMarkdown": false,
  "pdfExportSettings": {
    "pageSize": "Letter",
    "landscape": false,
    "margin": "0",
    "downscalePercent": 100
  },
  "readableLineLength": true,
  "mobilePullAction": "command-palette:open",
  "defaultViewMode": "source",
  "livePreview": true,
  "legacyEditor": false,
  "spellcheck": true,
  "spellcheckLanguages": [
    "en-US"
  ]
}
```

If this behavior is as expected, It seems difficult to implement because there are no APIs related to `spell checker/dictionary` in Obsidian API .

**Comment by chrisgrieser** [comment]
Hmm, for me they do show – maybe it takes a bit of time until the cache is written to that file or something? 🤔 
<img width="585" alt="Screenshot 2021-12-25 13 51 34" src="https://user-images.githubusercontent.com/73286100/147385254-08d8219c-7669-4dd6-b70b-87218c3a73f9.png">

**Comment by tadashi-aikawa** [comment]
I couldn't reproduce it even after reboot a machine. 🤔 (my machine is Windows?)

**Comment by chrisgrieser** [comment]
could be an Windows issue since on mac, Obsidian uses a different type of spellcheck. no idea though 🤔

**Comment by tadashi-aikawa** [comment]
Let me make this FR low priority because there are no interfaces on Obsidian API about this feature.

**Comment by chrisgrieser** [comment]
I just came back to this, and `app.vault.config.spellcheckDictionary` in the console gives me an array of all words I manually added. Wouldn't hurt adding it, since it's so easy to implement I guess? Like simply a toggle in the settings, ignoring it when for whatever reason the words are not saved in that config file for users of certain operating systems?

**Comment by tadashi-aikawa** [comment]
Ok. I'll implement it to v6.0.0-beta1. However, my windows machine has no `app.vault.config.spellcheckDictionary`, so I can't confirm it. Please test the feature after I release v6.0.0-beta1 🙏

**Comment by chrisgrieser** [comment]
Sure, thank You 🤗

**Comment by tadashi-aikawa** [comment]
> Please test the feature after I release v6.0.0-beta1 🙏

Oh.. I forgot to implement.., so it will release v6.0.0-beta3 😅

**Comment by tadashi-aikawa** [comment]
I found the dictionary file at `$env:APPDATA/obsidian/Custom Dictionary.txt`.

```
hoge
hogehoge
tadashi
tadashi-aikawa
checksum_v1 = 771945cfd7d9baa2348a00aa4f00a90c
```

In macOS, Is the dictionary file at `~/obsidian/Custom Dictionary.txt` by any chance?

**Comment by chrisgrieser** [comment]
> In macOS, Is the dictionary file at ~/obsidian/Custom Dictionary.txt by any chance?

Nope, they are in `.obsidian/app.json`, as an array with the key `spellcheckDictionary`

**Comment by chrisgrieser** [comment]
okay, i updated to beta3, where/how do I use the new feature? can't find anything in the settings and in the commands? 🤔

**Comment by chrisgrieser** [comment]
okay, there is some strange stuff with the spellcheck dictionary going on, see here: https://discord.com/channels/686053708261228577/716028884885307432/965367785507160064

**Comment by tadashi-aikawa** [comment]
Sorry for the confusion. It doesn't release yet 😅  (the release note is correct). I'll release it after solving your problem in #114 .

> okay, there is some strange stuff with the spellcheck dictionary going on, see here (and the following messages where I describe it): https://discord.com/channels/686053708261228577/716028884885307432/965367785507160064

Thanks! That would be very helpful. 😄

**Comment by tadashi-aikawa** [comment]
@chrisgrieser 
I have released [v6.0.0-beta7](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/releases/tag/6.0.0-beta7) 🚀 

Please confirm after enabling the below option whether it works as you expected.

![image](https://user-images.githubusercontent.com/9500018/165972240-77b77069-16d4-4edc-82f6-346ce74a3e09.png)

**Comment by tadashi-aikawa** [comment]
> .obsidian/app.json contains only the spellcheck words for the outdated legacy editor 

Oh... I overlooked `legacy editor` 😅  If so, there is nothing to do unless Obsidian API for spellcheck is published. 🤔
